vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1293 @@
+                    ### FIXME: make this contains(cons._v1, cplxIdx)  -- but what if we subtract on our way down?  need a peeled version
+                    if cons._v2.symtype == SYMT_CONST:
+                        symvar = cons._v1
+                        symcmp = cons._v2
+                    elif cons._v1.symtype == SYMT_CONST:
+                        symcmp = cons._v1
+                        symvar = cons._v2
+                    else:
+                        # neither side of the constraint is a CONST.  this constraint does 
+                        # not set static bounds on idx
+                        logger.debug("SKIPPING (non-discrete): cons = %s", repr(cons))
+                        continue
+
+                    if not contains(symvar, baseIdx):
+                        logger.debug("Constraint not based on Index: %r" % cons)
+                        continue
+                        '''
+                # there are two important offsets: constraint offsets and index offsets
+                #   index offsets are mostly subtractions from the actual number used (eg. index 1500 would be idx-1500 for an offset of 1500)
+                #   constraint offsets are subtractions from the index at the point of the constraint check.  these are accounted for by getNormalizedConstraints()
+                for con, stype, offset in self.getNormalizedConstraints():
+
+                    #conthing, consoff = peelIdxOffset(symvar)
+                    if stype == SYMT_CON_GT: # this is setting the lower bound
+                        newlower = offset + 1
+                        if newlower > lower: 
+                            logger.info("==setting a lower bound:  %s -> %s", lower, newlower)
+                            lower = newlower
+                        
+                    elif stype == SYMT_CON_GE: # this is setting the lower bound
+                        newlower = offset
+                        if newlower > lower: 
+                            logger.info("==setting a lower bound:  %s -> %s", lower, newlower)
+                            lower = newlower
+                        
+                    elif stype == SYMT_CON_LT: # this is setting the upper bound
+                        newupper = offset - 1
+                        if upper is None or newupper < upper and newupper > 0:
+                            logger.info("==setting a upper bound:  %s -> %s", upper, newupper)
+                            upper = newupper
+                        
+                    elif stype == SYMT_CON_LE: # this is setting the upper bound
+                        newupper = offset
+                        if upper is None or newupper < upper and newupper > 0:
+                            logger.info("==setting a upper bound:  %s -> %s", upper, newupper)
+                            upper = newupper
```

Comment:
I'm fairly certain we can just condense the repeated code here.

Reply:
if you like.  this is pretty key territory for bugs to crop up.  how would you like it changed?